KarhooUISDK/UIComponents/TermsConditionsView/TermsConditionsView.swift

```diff
@@ -18,56 +19,112 @@ public struct KHTermsConditionsViewID {
     public static let textView = "terms_text_view"
 }
 
+@available(*, deprecated, message: "Public access to this view will be removed in next release")
 public final class TermsConditionsView: UIView, UITextViewDelegate {
-    
-    private var didSetupConstraints = false
-    private var termsTextView: UITextView!
+
+    // MARK: - Nested type
+
+    private enum Mode {
+        case acceptRequired
+        case `default`
+    }
+
+    // MARK: - Properties
+
     public weak var delegate: TermsConditionsViewDelegate?
-    
-    init() {
-        super.init(frame: .zero)
-        self.setUpView()
+
+    public var isAccepted: Bool {
+        checkboxView.isSelected
     }
+
+    private let isAcceptanceRequired: Bool
+
+    // MARK: Views
+
+    private let termsTextView = UITextView().then {
+        $0.isAccessibilityElement = true
+        $0.translatesAutoresizingMaskIntoConstraints = false
+        $0.accessibilityIdentifier = KHTermsConditionsViewID.textView
+        $0.font = UIFont.systemFont(ofSize: 14.0)
+        $0.dataDetectorTypes = .link
+        $0.isSelectable = true
+        $0.isEditable = false
+        $0.isScrollEnabled = false
+        $0.tintColor = KarhooUI.colors.accent
+    }
+
+    private lazy var checkboxView = CheckboxView()
+
+    // MARK: - Lifecycle
     
-    override public func awakeFromNib() {
-        super.awakeFromNib()
+    init(isAcceptanceRequired: Bool = false) {
+        self.isAcceptanceRequired = isAcceptanceRequired
+        super.init(frame: .zero)
         self.setUpView()
     }
     
     required init?(coder: NSCoder) {
+        self.isAcceptanceRequired = false
         super.init(coder: coder)
         self.setUpView()
     }
-    
+
+    // MARK: - Setup
+
     private func setUpView() {
+        setupProperties()
+        setupHierarchy()
+        setupLayout()
+    }
+    
+    private func setupProperties() {
         translatesAutoresizingMaskIntoConstraints = false
         accessibilityIdentifier = KHTermsConditionsViewID.view
-        
-        termsTextView = UITextView()
-        termsTextView.isAccessibilityElement = true
-        termsTextView.translatesAutoresizingMaskIntoConstraints = false
-        termsTextView.accessibilityIdentifier = KHTermsConditionsViewID.textView
         termsTextView.delegate = self
-        termsTextView.font = UIFont.systemFont(ofSize: 14.0)
-        termsTextView.dataDetectorTypes = .link
-        termsTextView.textAlignment = .center
-        termsTextView.isSelectable = true
-        termsTextView.isEditable = false
-        termsTextView.isScrollEnabled = false
-        termsTextView.tintColor = KarhooUI.colors.accent
+    }
+
+    private func setupHierarchy() {
         addSubview(termsTextView)
-        
+        if isAcceptanceRequired { addSubview(checkboxView) }
+    }
+
+    private func setupLayout() {
         heightAnchor.constraint(greaterThanOrEqualToConstant: 10.0).isActive = true
```

Comment:
1. Didn't notice that. Maybe we should incorporate `greaterThanOrEqualToConstant` into our `UIView` extensions. Not as part of this PR though
2. Since it's a minimum, not fixed height I think we could go for 8 or 12 instead of 10 (would need to test the UI with both to see what looks better). But we can also leave it as is for now and circle back during the cleanup later on. Up to you
3. Ok, we'll circle back to this later